### DISCUSSION TITLE
1.30.2

### DISCUSSION THREAD

**Discussion by dani-garcia** [dani-garcia/vaultwarden#4289] [General]
⚠️ **Note:** The WebSockets service for live sync has been integrated in the main HTTP server, which means simpler proxy setups that don't require a separate rule to redirect WS traffic to port 3012. Please check the updated examples in the [wiki](https://github.com/dani-garcia/vaultwarden/wiki/Proxy-examples). It's recommended to migrate to this new setup as using the old server on port 3012 is deprecated, won't receive new features and will be removed in the next release.

## What's Changed
* Prevent generating an error during ws close by @BlackDex in https://github.com/dani-garcia/vaultwarden/pull/4127
* Update Rust, Crates, Profile and Actions by @BlackDex in https://github.com/dani-garcia/vaultwarden/pull/4126
* Several small fixes for open issues by @BlackDex in https://github.com/dani-garcia/vaultwarden/pull/4143
* Fix the version string by @BlackDex in https://github.com/dani-garcia/vaultwarden/pull/4153
* Decrease JWT Refresh/Auth token by @BlackDex in https://github.com/dani-garcia/vaultwarden/pull/4163
* Update crates by @BlackDex in https://github.com/dani-garcia/vaultwarden/pull/4173
* Add additional build target which optimizes for size by @gladiac in https://github.com/dani-garcia/vaultwarden/pull/4096
* Update web-vault to v2023.12.0 by @BlackDex in https://github.com/dani-garcia/vaultwarden/pull/4201
* Update Rust and Crates by @BlackDex in https://github.com/dani-garcia/vaultwarden/pull/4211
* Fix Single Org Policy check by @BlackDex in https://github.com/dani-garcia/vaultwarden/pull/4207
* Allow customizing the featureStates by @PKizzle in https://github.com/dani-garcia/vaultwarden/pull/4168
* Fix #3413: push to users accessing the collections using groups by @matlink in https://github.com/dani-garcia/vaultwarden/pull/3757
* US or EU Data Region Selection by @toto-xoxo in https://github.com/dani-garcia/vaultwarden/pull/3752
* enforce 2FA policy on removal of second factor and login by @stefan0xC in https://github.com/dani-garcia/vaultwarden/pull/3803
* improve emergency access when not enabled by @stefan0xC in https://github.com/dani-garcia/vaultwarden/pull/4227
* Update crates and fix icon issue by @BlackDex in https://github.com/dani-garcia/vaultwarden/pull/4237
* Bump h2 from 0.3.23 to 0.3.24 by @dependabot in https://github.com/dani-garcia/vaultwarden/pull/4260
* Fix bulk collection deletion by @BlackDex in https://github.com/dani-garcia/vaultwarden/pull/4257
* fix: use black text for update badge (better contrast) by @tessus in https://github.com/dani-garcia/vaultwarden/pull/4245
* prevent side effects if groups are disabled by @stefan0xC in https://github.com/dani-garcia/vaultwarden/pull/4265
* Update crates, web-vault to 2024.1.2 and GHA by @BlackDex in https://github.com/dani-garcia/vaultwarden/pull/4275
* Return 404 when user public_key is empty by @Timshel in https://github.com/dani-garcia/vaultwarden/pull/4271
* Improve file limit handling by @dani-garcia in https://github.com/dani-garcia/vaultwarden/pull/4242
* Fix attachment upload size check by @BlackDex in https://github.com/dani-garcia/vaultwarden/pull/4282
* err on invalid feature flag by @stefan0xC in https://github.com/dani-garcia/vaultwarden/pull/4263
* register missing push devices at login by @stefan0xC in https://github.com/dani-garcia/vaultwarden/pull/3792
* Update env template file by @gzfrozen in https://github.com/dani-garcia/vaultwarden/pull/4276

## New Contributors
* @gladiac made their first contribution in https://github.com/dani-garcia/vaultwarden/pull/4096
* @PKizzle made their first contribution in https://github.com/dani-garcia/vaultwarden/pull/4168
* @matlink made their first contribution in https://github.com/dani-garcia/vaultwarden/pull/3757
* @toto-xoxo made their first contribution in https://github.com/dani-garcia/vaultwarden/pull/3752
* @Timshel made their first contribution in https://github.com/dani-garcia/vaultwarden/pull/4271
* @gzfrozen made their first contribution in https://github.com/dani-garcia/vaultwarden/pull/4276

**Full Changelog**: https://github.com/dani-garcia/vaultwarden/compare/1.30.1...1.30.2

<hr /><em>This discussion was created from the release <a href='https://github.com/dani-garcia/vaultwarden/releases/tag/1.30.2'>1.30.2</a>.</em>

**Comment by ashoktvm** [comment]
Does this update have passkeys login enabled for web login?

**Reply by PKizzle** [reply]
I guess you are talking about "passwordless-login". No, this is not yet supported as can be seen here:
> Yes, exactly. This is how Vaultwarden currently looks when enabling `passwordless-login` (the spinner never stops as the API call fails): ![Screenshot passwordless login](https://private-user-images.githubusercontent.com/39984529/290950436-9b5727ab-41f6-4869-b390-f5a12e9aa979.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDY3OTYxNjIsIm5iZiI6MTcwNjc5NTg2MiwicGF0aCI6Ii8zOTk4NDUyOS8yOTA5NTA0MzYtOWI1NzI3YWItNDFmNi00ODY5LWIzOTAtZjVhMTJlOWFhOTc5LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAyMDElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMjAxVDEzNTc0MlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWE3OTdhODA4NGUzOTE0YWVjOTI1MmU3MWVhZDYyMzU2ZDI5Nzc5MDU1MDlmODdlNmFkM2UxZjc1ODdjMzkxZmImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.ugg3EjmNqMFMEKVI5IEIXMDvxBBXZUgdMb0NNmStWV0)

For further discussions about this feature please have a look over [here](https://github.com/dani-garcia/vaultwarden/discussions/4236).
